packages/eslint-plugin-lit-a11y/lib/rules/interactive-supports-focus.js

```diff
@@ -0,0 +1,108 @@
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        'Elements with an interactive role and interaction handlers (mouse or key press) must be focusable.',
+      category: 'Fill me in',
+      recommended: false,
+    },
+    fixable: null, // or "code" or "whitespace"
+    schema: [
+      // fill in your schema
+    ],
+  },
+
+  create(context) {
+    // variables should be defined here
+
+    //----------------------------------------------------------------------
+    // Helpers
+    //----------------------------------------------------------------------
+
+    const tabbable = (context.options && context.options[0] && context.options[0].tabbable) || [];
+
+    const interactiveProps = [...eventHandlersByType.mouse, ...eventHandlersByType.keyboard];
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    return {
+      TaggedTemplateExpression: node => {
+        if (
+          node.type === 'TaggedTemplateExpression' &&
+          node.tag.type === 'Identifier' &&
+          node.tag.name === 'html'
+        ) {
+          // const type = elementType(node);
+
+          const analyzer = TemplateAnalyzer.create(node);
+
+          analyzer.traverse({
+            enterElement: element => {
+              const hasInteractiveProps = Object.keys(element.attribs).some(attr =>
+                interactiveProps.includes(attr),
+              );
+              const hasTabindex = getTabIndex(getAttrVal(element.attribs.tabindex)) !== undefined;
+
+              if (
+                !hasInteractiveProps &&
+                isDisabledElement(element.attribs) &&
+                isHiddenFromScreenReader(element.name, element.attribs) &&
+                isPresentationRole(element.attribs)
+              ) {
+                // Presentation is an intentional signal from the author that this
+                // element is not meant to be perceivable. For example, a click screen
+                // to close a dialog.
+                return;
+              }
+
+              if (
+                hasInteractiveProps &&
+                // && isInteractiveRole(element.name, attributes)
+                !isInteractiveElement(element.name, element.attribs) &&
+                // && !isNonInteractiveElement(element.name, attributes)
+                // && !isNonInteractiveRole(element.name, attributes)
+                !hasTabindex
+              ) {
+                const role = getAttrVal(element.attribs.role);
+                if (includes(tabbable, role)) {
+                  // Always tabbable, tabIndex = 0
+                  context.report({
+                    node,
+                    message: `Elements with the '${role}' interactive role must be tabbable.`,
```

Comment:
eslint has it's own interpolation feature which makes it easier to reuse messages.

this file is a good use case for messageId as well

take a look at the eslint docs:

https://eslint.org/docs/developer-guide/working-with-rules#using-message-placeholders